core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -339,6 +339,27 @@ private[chisel3] object ir {
       extends Definition
   case class DefObject(sourceInfo: SourceInfo, id: HasId, className: String) extends Definition
 
+  class Placeholder(val sourceInfo: SourceInfo) extends Command {
+    private var buffer: mutable.Builder[Command, ArraySeq[Command]] = null
```

Comment:
I don't know offhand, can you comment about this in comparison to what's used for the buffer in Block, for example?  Is this same but better/more explicit, or different functionally in an important way?
(`ArraySeq.newBuilder[Command]`)

Reply:
This is just the type that is returned by `ArraySeq.newBuilder[Command]` [[1]]. The PR is trying to be clever here and initialize this to `null` to lazily initialize this. (Maybe a `lazy val` would be better than manually doing this lazily.) In order to do that, the PR needs to set the type explicitly while also making it `null`.

[1]: https://www.scala-lang.org/api/2.13.x/scala/collection/immutable/ArraySeq$.html#newBuilder[A](implicitevidence$2:scala.reflect.ClassTag[A]):scala.collection.mutable.Builder[A,scala.collection.immutable.ArraySeq[A]]